src/fondant/compiler.py

```diff
@@ -183,3 +184,61 @@ def _generate_spec(self, pipeline: Pipeline, extra_volumes: list) -> dict:
             "version": "3.8",
             "services": services,
         }
+
+
+class KubeFlowCompiler(Compiler):
+    """Compiler that creates a Kubeflow pipeline spec from a pipeline."""
+
+    def _resolve_imports(self):
+        """Resolve imports for the Kubeflow compiler."""
+        try:
```

Comment:
I think you can just do 

if is_kfp_available():
    import kfp
    from kfp import dsl

since the function will check for those things for you

Reply:
We should no longer need `is_kfp_available` with this approach. This is very nice change, since you'll no longer need to install kfp when using the local runner.

I think we can just move this to `__init__()` instead of this special function though?

Reply:
Created a PR to remove obsolete packages

https://github.com/ml6team/fondant/pull/293